server/src/definitions/DefinitionUtils.ts

```diff
@@ -93,5 +93,5 @@ export function getYmlEntityNamePartWithoutClassName(text: string, position: Ent
         return null;
     }
     const parts = result.split('::');
-    return parts[parts.length - 1];
+    return parts[position === EntityPartPosition.END ? parts.length - 1 : 0];
```

Comment:
I… think I got part of this, but hard to be sure. What about a comment (and an update of the description of the parameter, to explain why `position` can now have this special-ish value, what it means and what will ensue)?
(Next step if I still don’t understand: ask for an example xD)